enhancements/ingress/route-secret-injection-for-external-certificate-management.md

```diff
@@ -0,0 +1,375 @@
+
+## Summary
+
+Currently, users of OpenShift cannot very easily integrate OpenShift Routes
+with third-party certificate management solutions like [cert-manager](https://github.com/cert-manager/cert-manager).
+And this is mainly due to the restriction on Routes API requiring the certificate
+data to be burned in the CR as opposed to having a reference. This is especially
+problematic when third-party solutions also manage the life cycle (create/renew/delete)
+of the generated certificates which OpenShift Routes does not support and requires
+manual intervention to manage certificate life cycle.
+
+This enhancement aims to provide a solution where OpenShift Routes can support
+integration with third-party certificate management solutions like cert-manager and
+avoid manual certificate management by the user which is more error prone.
+
+## Motivation
+
+OpenShift customers currently manually manage certificates for user workloads
+by updating OpenShift Routes with the updated certificate data during expiry/renew
+workflow. This is cumbersome activity if users have a huge number of workloads and
+is also error prone. 
+
+This enhancement adds better support to OpenShift Routes for third-party certificate
+management solutions like cert-manager.
+
+### User Stories
+
+* As an OpenShift cluster administrator, I want to be able to provide a certificate reference on 
+OpenShift Routes so that I can integrate with third-party certificate management solution
+
+* As an OpenShift cluster administrator, I want to use third-party solutions like cert-manager
+for certificate management of user workloads on OpenShift so that no manual process is
+required to renew expired certificates.
+
+* As an OpenShift cluster administrator, I want OpenShift Routes to support both manual/managed
+mode of operation for certification management so that I can switch between manual certificate
+management and third-party certificate management.
+
+* As an OpenShift engineer, I want to be able to update Route API so that I can integrate
+OpenShift Routes with third-party certificate management solutions like cert-manager.
+
+### Goals
+
+- Provide a user configurable option in Route API to provide externally managed certificate data.
+- Provide users a mechanism to switch between using externally managed certificates and
+  manually managing certificates on OpenShift routes.
+- Provide smooth roll out of new certificates on OpenShift router when managed certificates
+  are renewed.
+
+### Non-Goals
+
+- Provide certificate life cycle management controls on the Route API (expiryAfter, renewBefore, etc).
+- Provide migration of certificates from current solution.
+
+## Proposal
+
+This enhancement proposes introducing a new controller (secret-injector) in route-controller-manager
+to manage a new annotation on the Route CRD. This annotation enables the users to provide a reference to 
+a Secret containing the serving cert/key pair that will be served by OpenShift router.
```

Comment:
I think it's worth mentioning  mentioning that the contents of the secret will be added to `.spec.tls` fields.

Reply:
Addressed.